### PR TITLE
[e2e] Remove test redundancies

### LINKED ISSUE #1404: E2E tests redundancy
`main` e2e tests currently contains tests that can be considered redundant and add up to the total execution time while providing minimal to zero added value.

**TLS:**
* `TestTLSWithExistingKeystore`
* `TestClusterFormationWithTLS`
Difference between the two above is one running 1 replica while the other runs 2 replicas. The latter is a stronger case of the first one. If the first one fails then the second one fails too. Thus having `TestClusterFormationWithTLS` should enough.

**Graceful Shutdown**
* `TestPermanentCache`
* `TestCheckDataSurviveToShutdown`
The two tests are testing basically the same thing. One is verifying the cache exists and is usable after graceful shutdown when created with `PERNAMENT` (which is default for some time now). The other checks that the cache exists and data were not lost with persistence configured. In this case the later is again stronger case and there is no added value by first one.

**Basic**
* `TestNodeStartup`
* `TestClusterFormation`
While there is a subtle difference where the first one tests also label propagation it's not clearly communicated out in the test name and if the node startup fails then all the tests fails. This two could be merged in a single 'SmokeTest' with a single deployment that would execute subtests on that deployment (verify clustering, label and other possible defaults).

**External service**
* `TestExternalService`
* `TestExternalServiceWithAuth`
All the tests are exposing external service and thus test it. Added value in those tests may be explicit configuration of user secrets by the later one but the test should be renamed. Default user verification could be included in the `SmokeTest` deployment above.

### PR DESCRIPTION
**Pull request by Crumby**
* Resolves #1404

### CODE REVIEW

File: test/e2e/main/scaling_test.go [hunk 1]
```diff
@@ -9,79 +9,62 @@ import (
...
 
-	cacheName := "test"
-	// Define function for the generic stop/start test procedure
-	var createPermanentCache = func(ispn *ispnv1.Infinispan) {
-		client_ := tutils.HTTPClientForCluster(ispn, testKube)
-		tutils.NewCacheHelper(cacheName, client_).CreateWithDefault("PERMANENT")
-	}
+	// Cache definitions
+	defaultCache := "default-cache"
+	defaultCacheConfig := `<distributed-cache name="` + defaultCache + `" mode="SYNC" statistics="true">
```
Comment (ryanemerson) [inline test/e2e/main/scaling_test.go:20]: For ease of reading I think we should simplify this to just `<distributed-cache name=" + defaultCache + "/>"

File: test/e2e/main/scaling_test.go [hunk 2]
```diff
@@ -9,79 +9,62 @@ import (
...
+	defaultKey := "defaultKey"
+	defaultValue := "defaultValue"
 
-	// Delete the cluster
-	testKube.GracefulShutdownInfinispan(spec)
-	testKube.GracefulRestartInfinispan(spec, 1, tutils.SinglePodTimeout)
+	defaultCacheHelper.TestBasicUsage(defaultKey, defaultValue)
+
+	// Server may return 500 upon deleting cache after graceful shutdown of a cluster
+	// defaultCacheHelper.Delete()
```
Comment (ryanemerson) [inline test/e2e/main/scaling_test.go:60]: This doesn't sound right, any idea why a 500 response is being returned?

File: test/e2e/main/smoke_test.go [hunk 3]
```diff
@@ -18,83 +22,79 @@ import (
 func TestNodeStartup(t *testing.T) {
 	defer testKube.CleanNamespaceAndLogOnPanic(t, tutils.Namespace)
 
-	// Create a resource without passing any config
+	// Infinispan cluster defintion with extra labels to be propagated to the service and pods
+	replicas := 2
```
Comment (ryanemerson) [inline test/e2e/main/smoke_test.go:26]: I think we should keep the `TestClusterFormation` name instead of `TestNodeStarupt` as the former implies pod startup and cluster creation works.

File: test/e2e/main/scaling_test.go (hunk 2 above)
Comment (Crumby) [reply test/e2e/main/scaling_test.go:60]: No idea, it doesn't happen always but often enough. There is an issue reported already.

Between keeping `deleteCache` and increasing number of cluster nodes to two I came to a conclusion that the later is more important to be covered by tests.

Comment (ryanemerson) [reply test/e2e/main/scaling_test.go:60]: > There is an issue reported already.

Do you have a link? I can't find it

File: test/e2e/main/scaling_test.go (hunk 1 above)
Comment (Crumby) [reply test/e2e/main/scaling_test.go:20]: Unfortunately I'm not following up. Could you elaborate?

File: test/e2e/main/smoke_test.go (hunk 3 above)
Comment (Crumby) [reply test/e2e/main/smoke_test.go:26]: Could be, I was actually thinking on who to name this one as it doesn't test only cluster formation but default authentication and labels too. Maybe `TestDefaultClusterFeatures`?

File: test/e2e/main/scaling_test.go (hunk 1 above)
Comment (ryanemerson) [reply test/e2e/main/scaling_test.go:20]: I got the markdown wrong, causing the config to be hidden. Hopefully the original comment is a bit clearer now.

tldr; Lets remove the unnecessary attributes and elements from the cache definition to make the test as simple as possible.

File: test/e2e/main/scaling_test.go (hunk 2 above)
Comment (ryanemerson) [reply test/e2e/main/scaling_test.go:60]: [ISPN-13740](https://issues.redhat.com/browse/ISPN-13740)

File: test/e2e/main/smoke_test.go (hunk 3 above)
Comment (ryanemerson) [reply test/e2e/main/smoke_test.go:26]: Sounds good.

File: test/e2e/main/scaling_test.go (hunk 1 above)
Comment (Crumby) [reply test/e2e/main/scaling_test.go]: Replaced as suggested. Even though it would be nice to have option to define caches in some friendly way in the code. Using xml in any code is just painful. But so is that the cache doesn't have defaults for some features it nags you to have defined (encoding, fetch-state for persistent caches).

File: test/e2e/main/scaling_test.go (hunk 2 above)
Comment (Crumby) [reply test/e2e/main/scaling_test.go:59]: Added Jira ID to the comment

**Comment by Crumby** [comment]
@ryanemerson I also updated number of entries used by backup-restore tests. That can be technically considered redundancy too.

**Comment by ryanemerson** [comment]
Thanks @Crumby
